sdk/servicebus/Azure.Messaging.ServiceBus/tests/Diagnostics/EventSourceLiveTests.cs

```diff
@@ -48,7 +49,7 @@ await using (var scope = await ServiceBusScope.CreateWithQueue(enablePartitionin
                 _listener.SingleEventById(ServiceBusEventSource.CreateMessageBatchStartEvent, e => e.Payload.Contains(sender.Identifier));
                 _listener.SingleEventById(ServiceBusEventSource.CreateMessageBatchCompleteEvent, e => e.Payload.Contains(sender.Identifier));
 
-                IEnumerable<ServiceBusMessage> messages = ServiceBusTestUtilities.AddMessages(batch, messageCount).AsReadOnly<ServiceBusMessage>();
+                IEnumerable<AmqpMessage> messages = ServiceBusTestUtilities.AddMessages(batch, messageCount).AsReadOnly<AmqpMessage>();
```

Comment:
Suggest leaving this as is and just having a single test for this scenario.

Reply:
Same here.   Not holding onto `ServiceBusMessage` instances in the batch is kind of the entire point of this change.  😄

Reply:
I'm not sure how to leave this as is. The batch is only able to do AsReadOnly<AmqpMessage> with the new changes. Should I have written a way for the batch to be able to return SBmessages?

Reply:
Good point 😄 I would say we either add a test method that returns back the SBMessages or we just use the regular SendMessagesAsync overload in all of these tests.

Reply:
I think we can use this [method](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/tests/Infrastructure/ServiceBusTestUtilities.cs#L19) for any test that isn't really testing batches.

Reply:
So, in this case the test is explicitly testing batches, so we'd need to either add a new utility message that returns the SBMessages or just leave what you have. I'm fine with leaving as is for the tests that are specifically testing batches. I was just opposed to changing every test to use `AmqpMessage`.